src/overlays/actors/ovl_En_Dg/z_en_dg.c

```diff
@@ -980,52 +1149,62 @@ void func_8098B560(EnDg* this, PlayState* play) {
         this->actor.velocity.y = -0.5f;
     }
 
-    sp34 = BgCheck_EntityRaycastFloor2(play, &play->colCtx, &poly, &pos);
+    floorHeight = BgCheck_EntityRaycastFloor2(play, &play->colCtx, &poly, &pos);
+
+    // Checking *only* that the dog is touching a wall here can result in some strange
+    // behavior. For example, if the player throws the dog at a wall next to some water,
```

Comment:
```suggestion
    // Checking *only* that the dog is touching a wall that can result in some strange
    // behavior. For example, if the player throws the dog at a wall next to some water,
```

Reply:
I'm not sure your suggestion is grammatically correct? I don't quite know what you're going for here.

Reply:
No, that's not what I was going for. I was saying "**The fact that they are** checking *only* that the dog is touching the wall **in the below code** can result in strange behavior". I just shortened/omitted the bolded parts.

Reply:
I would need more info on what the code is doing for me to make a suggestion on how to eliminate the chunkiness here.

Given the hardcoded floorHeight, I assume this code is really only applicable specifically to the South Clocktown dog correct? What sort of floors have a floorHeight > -100 here?

Reply:
Yeah, the South Clock Town dog is the only dog that is even near water, so strictly by definition, it only applies to them in the final game. I tried cleaning this comment up a bit to explain things better.